packages/next/client/page-loader.js

```diff
@@ -107,30 +124,32 @@ export default class PageLoader {
     })
   }
 
-  async loadScript (route) {
+  async loadRouteBundle (route) {
     await this.promisedBuildId
 
     route = this.normalizeRoute(route)
     let scriptRoute = route === '/' ? '/index.js' : `${route}.js`
 
-    const script = document.createElement('script')
+    const url = `${this.assetPrefix}/_next/static/${encodeURIComponent(
+      this.buildId
+    )}/pages${scriptRoute}`
+    this.loadScript(url, route)
+  }
 
+  loadScript (url, route) {
+    const script = document.createElement('script')
     if (process.env.__NEXT_MODERN_BUILD && 'noModule' in script) {
       script.type = 'module'
-      scriptRoute = scriptRoute.replace(/\.js$/, '.module.js')
+      url = url.replace(/\.js$/, '.module.js')
     }
-
-    const url = `${this.assetPrefix}/_next/static/${encodeURIComponent(
-      this.buildId
-    )}/pages${scriptRoute}`
     script.crossOrigin = process.crossOrigin
     script.src = url
+    script.async = false // Explicit async = false required for in-order execution
```

Comment:
Webpack chunks don't require in-order execution, correct? We can probably remove this.